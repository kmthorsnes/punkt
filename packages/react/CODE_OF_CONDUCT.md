# Retningslinjer for å bidra

Her skriver vi vår code of conduct.