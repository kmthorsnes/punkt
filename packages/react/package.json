{
  "name": "@oslokommune/punkt-react",
  "version": "11.4.2",
  "description": "React komponentbibliotek til Punkt, et designsystem laget av Oslo Origo",
  "homepage": "https://punkt.oslo.kommune.no",
  "author": "Team Designsystem, Oslo Origo",
  "main": "dist/punkt-react.umd.js",
  "module": "dist/punkt-react.es.js",
  "files": [
    "/dist",
    "src/components/**/*",
    "CHANGELOG.md",
    "CONTRIBUTING.md"
  ],
  "type": "module",
  "types": "dist/index.d.ts",
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "import": "./dist/punkt-react.es.js",
      "require": "./dist/punkt-react.umd.js"
    }
  },
  "scripts": {
    "dev": "vite",
    "build": "tsc && vite build",
    "build-app": "tsc && vite build --config vite.config-app.ts",
    "lint": "eslint 'src/**/*.{js,jsx,ts,tsx}'",
    "lint:fix": "eslint --fix 'src/**/*.{jsx,ts,tsx}'",
    "format": "prettier --write src//**/*.{ts,tsx,css} --config ./.prettierrc",
    "preview": "vite preview --outDir dist-app",
    "test": "react-scripts test --env=jsdom"
  },
  "devDependencies": {
    "@oslokommune/punkt-assets": "^11.0.0",
    "@oslokommune/punkt-css": "^11.4.1",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "^14.4.3",
    "@types/jest": "^29.5.3",
    "@types/jest-axe": "^3.5.5",
    "@types/node": "^18.16.19",
    "@types/react": "^18.2.14",
    "@types/react-dom": "^18.2.6",
    "@vitejs/plugin-react": "^3.1.0",
    "classnames": "^2.3.2",
    "eslint": "^8.56.0",
    "eslint-config-prettier": "^9.1.0",
    "eslint-plugin-prettier": "^5.1.0",
    "eslint-plugin-simple-import-sort": "^10.0.0",
    "jest": "^29.6.1",
    "jest-axe": "^8.0.0",
    "jest-environment-jsdom": "^29.6.1",
    "prettier": "^3.1.1",
    "react-hooks": "^1.0.1",
    "react-router-dom": "^6.14.1",
    "react-scripts": "^5.0.1",
    "sass": "^1.63.6",
    "typescript": "^4.9.3",
    "vite": "^4.4.12",
    "vite-plugin-dts": "^3.6.4"
  },
  "peerDependencies": {
    "@oslokommune/punkt-assets": ">=3.0.0",
    "@oslokommune/punkt-css": ">=3.0.0",
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  },
  "engines": {
    "node": ">=16.0.0"
  },
  "private": false,
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "keywords": [
    "punkt",
    "designsystem",
    "react",
    "reactjs",
    "components",
    "ux",
    "ui"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/oslokommune/punkt.git"
  },
  "bugs": {
    "url": "https://github.com/oslokommune/punkt/issues"
  },
  "license": "MIT"
}
