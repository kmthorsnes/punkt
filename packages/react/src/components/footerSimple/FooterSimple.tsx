import React from 'react';

import { PktIcon } from '../icon/Icon';


export interface IPktFooterSimple extends React.HTMLAttributes<HTMLDivElement> {
  links?: Array<{
    href: string;
    text: string;
    external?: boolean;
    openInNewTab?: boolean;
  }>;
  personvernOgInfoLink?: string
  tilgjengelighetLink?: string
}

export const PktFooterSimple: React.FC<IPktFooterSimple> = ({
  links = [],
  personvernOgInfoLink = "https://www.oslo.kommune.no/personvern-og-informasjonskapsler/",
  tilgjengelighetLink = "https://www.oslo.kommune.no/tilgjengelighet/",
  className
}) => {
  const classNames = [className, 'pkt-footer-simple'].filter(Boolean).join(' ')
  return (
    <footer className={classNames} data-mode="dark">
      <div className="pkt-footer-simple__container">
        <ul className="pkt-footer-simple__list">
          {links.map((link, index) => (
            <li className="pkt-footer-simple__list-item" key={index}>
              <a
                className={`pkt-footer-simple__link ${link.external ? ' pkt-link--external' : ''}`}
                href={link.href}
                target={link.openInNewTab ? '_blank' : '_self'}
                rel={link.openInNewTab ? 'noopener noreferrer' : undefined}
              >
                <PktIcon
                  className="pkt-footer-simple__link-icon"
                  name="chevron-right"
                />
                {link.text}
              </a>
            </li>
          ))}
          <li className="pkt-footer-simple__list-item">
            <a className="pkt-footer-simple__link" href={personvernOgInfoLink}>
              <PktIcon className="pkt-footer-simple__link-icon" name="chevron-right"></PktIcon>
              Personvern og informasjonskapsler
            </a>
          </li>
          <li className="pkt-footer-simple__list-item">
            <a className="pkt-footer-simple__link" href={tilgjengelighetLink}>
              <PktIcon className="pkt-footer-simple__link-icon" name="chevron-right"></PktIcon>
              Tilgjengelighet
            </a>
          </li>
        </ul>
      </div>
    </footer>
  );
};
