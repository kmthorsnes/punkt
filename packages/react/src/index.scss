@use 'sass:map';
@use 'pkt/scss/abstracts/variables' with ($font-path: 'pktAssets'
);
@use 'pkt/scss/abstracts/functions';
@use 'pkt/scss/abstracts/mixins/breakpoints' as *;
@use 'pkt/scss/abstracts/mixins/typography';
@use 'pkt/scss/pkt.scss';

body {
  background-color: #efefef;
  margin: 0;
  min-width: 360px;
  padding: 0;
}

h1 {
  @include typography.get-text('pkt-txt-28');
  margin-bottom: map.get(variables.$spacing, 'size-20');

  @include bp('tablet-up') {
    @include typography.get-text('pkt-txt-36');
  }
}

h2 {
  @include typography.get-text('pkt-txt-24');
  margin-bottom: map.get(variables.$spacing, 'size-10');

  @include bp('tablet-up') {
    @include typography.get-text('pkt-txt-30');
  }
}

p {
  margin-bottom: map.get(variables.$spacing, 'size-15');
}

.pkt-logo {
  width: 50%;
  height: 50%;
  max-width: 18rem;
  display: inline-block;
  margin: 1rem;
}

.page-container {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: auto 1fr;
  grid-column-gap: 0;
  grid-row-gap: 0;
  grid-template-areas:
    'header'
    'main';
}

.dev-header {
  $-header-height: 50px;
  $-header-height-phablet-up: 60px;
  $-logo-width: calc($-header-height * 60 / 31);
  $-logo-width-phablet-up: calc($-header-height-phablet-up * 60 / 31);

  background-color: var(--pkt-color-background-default);
  border-bottom: 1px solid var(--pkt-color-border-subtle);
  display: grid;
  gap: 0 0;
  grid-area: header;
  grid-template-areas: 'dev-header-logo . dev-header-nav';
  grid-template-columns: minmax($-logo-width, $-logo-width-phablet-up) 1fr auto auto;
  padding: map.get(variables.$spacing, 'size-20') map.get(variables.$spacing, 'size-15');
  align-items: center;

  &__logo {
    grid-area: dev-header-logo;

    &-link {
      display: block;
    }

    &-svg {
      height: $-header-height;
      width: calc($-header-height * 60 / 31);

      @include bp('phablet-up') {
        height: $-header-height-phablet-up;
        width: calc($-header-height-phablet-up * 60 / 31);
      }
    }
  }

  &__nav {
    grid-area: dev-header-nav;
  }
}

.component {
  background-color: #fff;
  margin-bottom: map.get(variables.$spacing, 'size-40');
  padding: 10px;
  transition: all 0.3s ease-out;

  [data-mode='dark'] & {
    background-color: var(--pkt-color-brand-dark-blue-1000);
    color: #fff;
  }

  &__example {
    align-items: flex-start;
    display: flex;
    flex-wrap: wrap;
    gap: 5px 5px;
  }

  &__block-example>div {
    margin-bottom: 20px;
  }

  &__example+&__example,
  &__block-example+&__block-example {
    margin-top: map.get(variables.$spacing, 'size-5');
  }
}

$-checkbox-img: url("data:image/svg+xml,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 20 20'><path fill='none' stroke='#fff' stroke-linecap='round' stroke-linejoin='round' stroke-width='3' d='M6 10l3 3l6-6'/></svg>") !default;

.page-main {
  position: relative;
  grid-area: main;
  padding-bottom: map.get(variables.$spacing, 'size-15');
  padding-top: map.get(variables.$spacing, 'size-15');

  @include bp('tablet-up') {
    padding-bottom: map.get(variables.$spacing, 'size-30');
    padding-top: map.get(variables.$spacing, 'size-30');
  }

  &__toolbar {
    display: flex;
    position: absolute;
    right: 2rem;
    top: 2.25rem;
    justify-content: flex-end;
  }

  &__checkbox {
    appearance: none;
    background-color: var(--pkt-color-input-background-normal);
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
    border: 2px solid var(--pkt-color-input-border-normal);
    display: inline-block;
    margin: 0.125rem 0.25rem 0 0;
    width: 1.25rem;
    height: 1.25rem;
    transition: all 0.3s ease-out;
    vertical-align: top;

    &[type='checkbox'] {
      border-radius: 0;
    }

    &:checked {
      &[type='checkbox'] {
        background-color: var(--pkt-color-input-border-normal);
        background-image: functions.escape-svg($-checkbox-img);
        border-color: var(--pkt-color-input-border-normal);
      }
    }
  }
}
