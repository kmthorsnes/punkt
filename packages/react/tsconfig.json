{
  "compilerOptions": {
    "target": "ESNext",
    "useDefineForClassFields": true,
    "lib": [
      "DOM",
      "DOM.Iterable",
      "ESNext"
    ],
    "allowJs": false,
    "skipLibCheck": true,
    "esModuleInterop": false,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "module": "ESNext",
    "moduleResolution": "Node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx",
    "baseUrl": ".",
    "paths": {
      "@/*": [
        "src/*"
      ],
      "pkt/*": [
        "node_modules/@oslokommune/punkt-css/dist/*"
      ],
      "pktAssets/*": [
        "node_modules/@oslokommune/punkt-assets/dist/*"
      ],
      "componentSpecs/*": [
        "../../component-specs/*"
      ],
      "punkt-react": [
        "src/components/index.ts"
      ]
    },
  },
  "include": [
    "src"
  ],
  "references": [
    {
      "path": "./tsconfig.node.json"
    }
  ]
}