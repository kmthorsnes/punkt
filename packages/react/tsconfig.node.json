{
  "compilerOptions": {
    "composite": true,
    "module": "ESNext",
    "moduleResolution": "Node",
    "allowSyntheticDefaultImports": true,
    "resolveJsonModule": true,
    "types": [
      "jest",
      "node"
    ]
  },
  "include": [
    "vite.config.ts",
    "package.json"
  ]
}