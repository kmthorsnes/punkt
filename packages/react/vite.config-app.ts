import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react'
import { resolve } from 'path'

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [react()],
  resolve: {
    alias: {
      "@": resolve(__dirname, "src"),
      "pkt": "@oslokommune/punkt-css/dist",
      "pktAssets": "@oslokommune/punkt-assets/dist",
      "componentSpecs": resolve(__dirname, "../../component-specs/"),
    },
  },
  build: {
    outDir: "./dist-app",
  },
});
