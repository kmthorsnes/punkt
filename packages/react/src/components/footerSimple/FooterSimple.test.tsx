// Test med axe https://www.npmjs.com/package/jest-axe
import '@testing-library/jest-dom/extend-expect';

import { render } from '@testing-library/react'
import { axe, toHaveNoViolations } from 'jest-axe'

import { PktFooterSimple } from './FooterSimple'

expect.extend(toHaveNoViolations)



describe('PktFooterSimple', () => {

  const links = [
    { href: 'https://www.example.com', text: 'Example Link 1' },
    { href: 'https://www.example.com', text: 'Example Link 2' },
    { href: 'https://www.example.com', text: 'Example Link 3' },
  ];

  describe('basic rendering', () => {
    it('renders a list of links', () => {
      const { getByText } = render(<PktFooterSimple links={links} />);
      links.forEach((link) => {
        expect(getByText(link.text)).toBeInTheDocument();
        expect(getByText(link.text).getAttribute('href')).toEqual(link.href);
      });
    });

    it('renders an external link with an external class', () => {
      const links = [
        { href: 'https://www.example.com', text: 'Example Link', external: true },
      ];
      const { getByText } = render(<PktFooterSimple links={links} />);
      expect(getByText(links[0].text)).toHaveClass('pkt-link--external');
    });

    it('renders a privacy link', () => {
      const { getByText } = render(<PktFooterSimple />);
      expect(getByText('Personvern og informasjonskapsler')).toBeInTheDocument();
      expect(getByText('Personvern og informasjonskapsler').getAttribute('href')).toEqual('https://www.oslo.kommune.no/personvern-og-informasjonskapsler/');
    });

    it('renders an accessibility link', () => {
      const { getByText } = render(<PktFooterSimple />);
      expect(getByText('Tilgjengelighet')).toBeInTheDocument();
      expect(getByText('Tilgjengelighet').getAttribute('href')).toEqual('https://www.oslo.kommune.no/tilgjengelighet/');
    });
  })

  describe('accessibility', () => {
    it('renders with no wcag errors with axe', async () => {
      const { container } = render(<PktFooterSimple links={links} />);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
    })
  })

  describe('snapshot', () => {
    it('renders correctly with all props', () => {
      const { container } = render(<PktFooterSimple links={links} />);
      expect(container).toMatchSnapshot();
    })
  })
});
