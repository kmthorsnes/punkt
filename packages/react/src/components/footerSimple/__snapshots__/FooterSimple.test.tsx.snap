// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PktFooterSimple snapshot renders correctly with all props 1`] = `
<div>
  <footer
    class="pkt-footer-simple"
    data-mode="dark"
  >
    <div
      class="pkt-footer-simple__container"
    >
      <ul
        class="pkt-footer-simple__list"
      >
        <li
          class="pkt-footer-simple__list-item"
        >
          <a
            class="pkt-footer-simple__link "
            href="https://www.example.com"
            target="_self"
          >
            Example Link 1
          </a>
        </li>
        <li
          class="pkt-footer-simple__list-item"
        >
          <a
            class="pkt-footer-simple__link "
            href="https://www.example.com"
            target="_self"
          >
            Example Link 2
          </a>
        </li>
        <li
          class="pkt-footer-simple__list-item"
        >
          <a
            class="pkt-footer-simple__link "
            href="https://www.example.com"
            target="_self"
          >
            Example Link 3
          </a>
        </li>
        <li
          class="pkt-footer-simple__list-item"
        >
          <a
            class="pkt-footer-simple__link"
            href="https://www.oslo.kommune.no/personvern-og-informasjonskapsler/"
          >
            Personvern og informasjonskapsler
          </a>
        </li>
        <li
          class="pkt-footer-simple__list-item"
        >
          <a
            class="pkt-footer-simple__link"
            href="https://www.oslo.kommune.no/tilgjengelighet/"
          >
            Tilgjengelighet
          </a>
        </li>
      </ul>
    </div>
  </footer>
</div>
`;
